### ISSUE TITLE
Spurious PLE1507 error: Invalid type for initial `os.getenv` argument

### ISSUE THREAD

**Issue by DimitriPapadopoulos** [astral-sh/ruff#6537]
* A minimal code snippet that reproduces the bug:
   ```py
  import os
  
  
  def f(using_clear_path):
      return os.getenv("PATH_TEST" if using_clear_path else "PATH_ORIG")
  ```
* The command you invoked:
  ```
  ruff --select PLE file.py
  ```
* The current Ruff settings: no `pyproject.toml` file
* The current Ruff version (`ruff --version`):
  ```
   ruff 0.0.284
  ```

**Comment by charliermarsh** [comment]
Thanks, fixing...
